features/polyExplorer/src/components/entityList/entityList.jsx

```diff
@@ -58,19 +17,26 @@ function getStartGroups(entityGroups) {
     return keys.pop();
 }
 
-const EntityList = () => {
-    const { entities, globalData, activeFilters, handleRemoveFilter } =
-        useContext(ExplorerContext);
-    const onRemoveFilter = handleRemoveFilter;
-    const filteredEntities = activeFilters.apply(entities);
-    const entityGroups = groupEntities(filteredEntities);
+export default function EntityList({ entityGroups }) {
     const allKeys = Object.keys(entityGroups);
     const [loadedEntities, setLoadedEntities] = useState({});
     const [toLoadKeys, setToLoadKeys] = useState(allKeys);
     const [hasMore, setHasMore] = useState(true);
     const listRef = useRef();
 
-    const handleLoadMoreData = () => {
+    function loadEntities() {
+        const loadedEntities = {};
+        const startGroups = getStartGroups(entityGroups);
+        for (let i = 0; i <= allKeys.indexOf(startGroups); i++) {
```

Comment:
Maybe use map or list comprehension here?

Reply:
OK I think I'll give up and refactor this function as well :D Originally I wanted to leave the original logic alone, but why not :P

Reply:
Done - though I kept the _logic_ intact as it was.